File: ProjectPO_Java/src/PO_java/LifeSymulator/Plants/DeadlyNightshade.java
package PO_java.LifeSymulator.Plants;

import PO_java.LifeSymulator.Organism;
import PO_java.LifeSymulator.Plant;
import PO_java.LifeSymulator.Tools.Cords;
import PO_java.LifeSymulator.World;

public class DeadlyNightshade extends Plant {
    public DeadlyNightshade(World world, Cords cords)
    {
        strength = 99;
        age = 0;
        sign = '%';
        name = "DeadlyNightshade";
        toKill = false;
        isDead = false;
        this.cords = cords;
        this.world = world;
    }

    @Override
    public void collision(Organism occupyingOrganism){
        world.addOrganismToKill(this);
        world.addOrganismToKill(occupyingOrganism);
        world.cleanBoardField(this.cords);
        world.cleanBoardField(occupyingOrganism.getCords());
        System.out.println(this.name + " killed " + occupyingOrganism.getName() + " while they was trying to eat it");
    }
}

File: ProjectCppPO/DeadlyNightshade.cpp
#include "DeadlyNightshade.h"

DeadlyNightshade::DeadlyNightshade(World* world, Cords cords)
{
	features.strength = 99;
	features.age = 0;
	features.look = '%';
	features.name = "Deadly Nightshade";
	features.toKill = false;
	location = cords;
	this->world = world;
}

//Metoda nadpisuj¹ca metodê z klasy rodzica ¿eby wilcze jagody zabija³y tego kto je zje
void DeadlyNightshade::collision(Organism* occupyingOrganism)
{
	world->addOrganismToKill(this);
	world->addOrganismToKill(occupyingOrganism);
	world->cleanBoardField(location);
	world->cleanBoardField(occupyingOrganism->getLocation());
	//dodanie raportu
	string report = features.name + " killed " + occupyingOrganism->getFeatures().name + " while they was eating it";
	world->addReport(report);
}
